File: types/life-event.ts
import type { Service } from "@/types/service";
export interface LifeTask { id:number; title:string; description:string; why_it_matters:string; task_type:string; timing_label:string; deadline_days:number|null; estimated_minutes:number|null; required_documents:string[]; official_url:string|null; official_source:string|null; caution:string|null; display_order:number; services:Service[]; }
export interface LifeEventSummary { id:number; slug:string; title:string; short_label:string; description:string; icon:string; category:string; audience:string[]; reviewed_at:string|null; task_count:number; }
export interface LifeEventDetail extends LifeEventSummary { tasks:LifeTask[]; }
export interface LifePlan { plan_id:number; life_event_id:number; status:string; completed_tasks:number; total_tasks:number; progress_percent:number; }

File: apps/api/app/schemas/life_event.py
from datetime import datetime

from pydantic import BaseModel, ConfigDict, Field

from app.schemas.service import ServiceResponse


class LifeTaskResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)
    id: int
    title: str
    description: str
    why_it_matters: str
    task_type: str
    timing_label: str
    deadline_days: int | None
    estimated_minutes: int | None
    required_documents: list[str]
    official_url: str | None
    official_source: str | None
    caution: str | None
    display_order: int
    services: list[ServiceResponse] = Field(default_factory=list)


class LifeEventSummary(BaseModel):
    model_config = ConfigDict(from_attributes=True)
    id: int
    slug: str
    title: str
    short_label: str
    description: str
    icon: str
    category: str
    audience: list[str]
    reviewed_at: datetime | None
    task_count: int = 0


class LifeEventDetail(LifeEventSummary):
    tasks: list[LifeTaskResponse]


class LifePlanResponse(BaseModel):
    plan_id: int
    life_event_id: int
    status: str
    completed_tasks: int
    total_tasks: int
    progress_percent: int


class TaskProgressUpdate(BaseModel):
    is_completed: bool
    reminder_at: datetime | None = None
    note: str | None = Field(default=None, max_length=500)
